jschon/jsonpointer.py

```diff
@@ -282,7 +292,14 @@ def __new__(
             if not (match := RelativeJSONPointer._regex.fullmatch(value)):
                 raise RelativeJSONPointerError(f"'{value}' is not a valid relative JSON pointer")
 
-            up, ref = match.group('up', 'ref')
+            up, over, ref = match.group('up', 'over', 'ref')
+
+        if over in (0, ''):
```

Comment:
I'd suggest `if not over` which succinctly passes `None`, `0` or `''`.

Declaring the parameter as `over: int = None` might concur more strictly with the spec, but I think this would be overkill since, in the context of index manipulation, `None` implies `0`. In any case, `if not over` allows the caller to obtain the `over` value from a `None`, `0` or `''`, which are essentially equivalent.

Reply:
Yes, all of that makes sense (and I like how it makes the test code more straightforward as well).

I originally had it as `over: int = None` but the way I was handling it got very fiddly.

Would it make senes to use `over: int = ''` because that's how things will come in by default from parsing as seen in the test case?  It just felt weird to do that.  `over: int = None` seemed OK on the principle that something can be type-or-`None`, but type-or-other-type seemed too weird.  But I don't really understand type hinting and what constitutes a reasonable vs unreasonable violation of the hint.

Reply:
I did want to allow `0` as a passed value, because logically it is supported, it is just not _visible_ in the string representation of the pointer.  And that way if you're doing math to calculate the adjustment you don't have to worry about converting `0` to `None` (or `''`).

Reply:
In hindsight allowing strings to be passed as kwargs was a mistake. It complicates the code and presents a confusing interface to callers - the caller can already pass the whole rjp string via `value`, so why would they want to split that up and pass component strings via kwargs? In fact, the only reason this was done was to permit laziness in the unit test. I'm going to proceed with merging the PR, but will create a subsequent commit to resolve this.

Reply:
@marksparkza yeah that makes sense to me!